File: companies/stripe/NextServerNumber.py
class NextServerNumber:
    def nextNumber(self, nums):
        if not nums:
            return 1
        nums_set = set()

        for n in nums:
            nums_set.add(n)
        
        max_num = max(nums_set)
        for i in range(1, max_num + 1):
            if i not in nums_set:
                return i
        
        return max_num + 1

def main():
    sol = NextServerNumber()
    print(sol.nextNumber([])) #1
    print(sol.nextNumber([1, 2, 5])) # 3
    print(sol.nextNumber([2, 3]))

if __name__ == '__main__':
    main()

File: companies/stripe/next_server_number.rb
# frozen_string_literal: true

require 'set'

class NextServerNumber
  def next_number(nums: [])
    return 1 if nums.empty?

    nums_set = Set.new(nums)
    max_num = nums_set.max

    (1..max_num).to_a.each do |num|
      return num unless nums_set.member?(num)
    end

    max_num + 1
  end
end